packages/prisma/schema.prisma

```diff
@@ -1267,3 +1267,23 @@ model SecondaryEmail {
   @@unique([userId, email])
   @@index([userId])
 }
+
+// Needed to store tasks that need to be processed by a background worker or Tasker
+model Task {
+  id          Int       @id @default(autoincrement())
+  createdAt   DateTime  @default(now())
+  updatedAt   DateTime  @updatedAt
+  // The time at which the task should be executed
+  scheduledAt DateTime  @default(now())
+  // The time at which the task was successfully executed
+  succeededAt DateTime?
+  // The task type to be executed. Left it as a freeform string to avoid more migrations for now. Will be enforced at type level.
+  type        String
```

Comment:
Will this also handle versioning of the types? 

Many times when building these types of systems, we want to version the tasks so that when they are processed, they are processed by a handler that knows the version. 

Reason is when the system has immense amount of traffic and there can be no down time, migrating to a v2 handler, let’s say, needs to work seamlessly and all v1 tasks need to be processed correctly as well.

Reply:
I don't see why not. We can follow up a naming convention like sendEmailV2, sendEmailV3 and so on and keep the handlers around for long as needed.